src/main/java/it/pagopa/pn/bff/mappers/publickeys/PublicKeysCheckIssuerStatusMapper.java

```diff
@@ -0,0 +1,28 @@
+package it.pagopa.pn.bff.mappers.publickeys;
+
+import it.pagopa.pn.bff.generated.openapi.msclient.user_attributes.model.Consent;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.apikeys.BffPublicKeysCheckIssuerResponse;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.apikeys.PublicKeysIssuerResponse;
+import org.mapstruct.Context;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * Mapstruct mapper interface, used to map the PublicKeysIssuerResponse
+ * to the BffPublicKeysIssuerResponse
+ */
+@Mapper
+public interface PublicKeysCheckIssuerStatusMapper {
+    PublicKeysCheckIssuerStatusMapper modelMapper = Mappers.getMapper(PublicKeysCheckIssuerStatusMapper.class);
+
+    default BffPublicKeysCheckIssuerResponse mapPublicKeysIssuerStatus(PublicKeysIssuerResponse publicKeysIssuerResponse, @Context Consent consent) {
+        BffPublicKeysCheckIssuerResponse bffPublicKeysCheckIssuerResponse = new BffPublicKeysCheckIssuerResponse();
+        bffPublicKeysCheckIssuerResponse.setIssuer(publicKeysIssuerResponse);
+        if(consent != null)
+            bffPublicKeysCheckIssuerResponse.setTosAccepted(consent.getAccepted());
+        else
+            bffPublicKeysCheckIssuerResponse.setTosAccepted(null);
+
+        return bffPublicKeysCheckIssuerResponse;
+    }
+}
```

Comment:
we can avoid to define the PublicKeysIssuerStatusMapper and we can do all the transformations here. Follow this steps:

1. remove the default and the body from mapPublicKeysCheckIssuerStatus
2. remove the `@Context`
3. use the `@Mapping` annotation (check map struct guide)
4. use the qualifiedByName and `@Named` annotation to map the tosAccepted key (check map struct guide)

TIPS: with `@Mapping` we can map a source object into a target object. With this we can map the micro-service publicKeysIssuerResponse into issuer. With `@Mapping`, qualifiedByName and `@Named` we can do the calculation to map the micro-service consent into tosAccepted.